uax29/Decoder.cs

```diff
@@ -0,0 +1,54 @@
+namespace uax29;
+
+using System.Buffers;
+using System.Text;
+
+/// A bitmap of Unicode categories
+using Property = uint;
+
+internal interface IDecoder<TSpan>
+{
+	static abstract Property Ignore { get; }

```

Comment:
Intuitively, a Decoder only needs to have DecodeFirstRune and DecodeLastRune. There are only two of them — one for char and one for byte, regardless of words/graphemes/sentences. In other words, Decoder is scoped/parameterized on TSpan only.

In turn, Dict and Ignore are scoped/parameterized to words/graphemes/sentences. Feels like these two ideas need not live in a single interface?

Reply:
Yeah, implementation detail.  I did this iteratively, so just sorta slapped new stuff on as needed.  It could be split out into different types at the cost of more generic arguments on various methods/types.